### PR TITLE
fix: OrgUnit valueType validation

### PR DESCRIPTION
**Pull request by abyot**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHookTest.java [hunk 1]
```diff
@@ -743,6 +745,8 @@ void successValidationDataElementOptionValueIsValid()
 
         hook.validateEvent( reporter, event );
 
+        System.out.println( " the errors:  " + reporter.getReportList().toString() );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHookTest.java:748]: Remove debug code.

**Review by stian-sandvold — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/OrgUnitValueTypeSupplier.java [hunk 2]
```diff
@@ -87,4 +99,15 @@ private void collectResourceIds( List<String> orgUnitAttributes, List<String> or
             }
         } );
     }
+
+    private void collectResourceIds( List<String> orgUnitDataElements, List<String> orgUnitIds,
+        Set<DataValue> dataElements )
+    {
+        dataElements.forEach( de -> {
+            if ( orgUnitDataElements.contains( de.getDataElement() ) && !StringUtils.isEmpty( de.getValue() ) )
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/OrgUnitValueTypeSupplier.java:107]: Why don't we already filter by type and query only for the org unit ids?

Comment (abyot) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/OrgUnitValueTypeSupplier.java:107]: orgUnitDataElements is already filtered one. it contains all dataElements of ou value type. ~~dataElements~~ > dataValues.

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHookTest.java (hunk 1 above)
Comment (abyot) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/EventDataValuesValidationHookTest.java]: done

**Review by enricocolasante — APPROVED** [review]

**Pull request merged**